src/modality-specific-files/task-events.md

```diff
@@ -177,8 +177,38 @@ The stimulus files can be added in a `/stimuli` directory
 `stim_file` column in `events.tsv` mentioning which stimulus file was used
 for a given event,
 
-There are no restrictions on the file formats of the stimuli files,
-but they should be stored in the `/stimuli` directory.
+It is RECOMMENDED to include a `stimuli.tsv` table containing a list of the stimuli present in the `/stimuli` directory.
+The `stimuli.tsv` would list the stimulus files as rows and provide more information about the stimulus files as columns, such as their types (for example, still images, movies, audio, multi-channel audio, etc), text descriptions, and optional columns including references, relevant HED tags, etc.
+Annotations for the still stimulus files can be properly held in `stimuli.tsv` as specific columns. The `stimuli.tsv` file can be accompanied by a `stimuli.json` file describing the columns in detail.
+The `stimuli.tsv` MUST include the following columns:
+
+{{ MACROS___make_columns_table(
+   {
+      "stim_file": ("REQUIRED", "There are no restrictions on the file formats of the stimuli files, but they should be stored in the `/stimuli` directory (under the root directory of the dataset; with OPTIONAL subdirectories). The values under the stim_file column correspond to a path relative to `/stimuli`. For example `images/cat03.jpg` will be translated to `/stimuli/images/cat03.jpg`. This column MUST appear as the first column in the file."),
+      "type": ("REQUIRED", "String. One of `movie`, `still_image`, `audio`, `annotation`, or Freeform.")
+   }
+) }}
+
+
+### Time-varying stimuli
+A single-line annotation in a `stimuli.tsv` column would be sufficient for the still stimulus files (for example, images). However, a time-varying stimulus file could need separate *temporal annotations* for every frame (that is, the smallest temporal resolution of the stimulus file).
+
+The temporal annotations for time-varying stimuli can be stored in a separate file, `stimulus.tsv`, which would hold the annotations for timepoints of the time-varying stimulus similar to an `events.tsv` file. `stimulus.tsv` MUST follow the same format and REQUIRED and OPTIONAL columns of the `events.tsv` file MAY be accompanied by a JSON file describing the columns in detail.
+
```

Comment:
A naming condition is needed here for the stimulus file.  There will be multiple stimulus files and they should be associated with their corresponding "movies" by name.  Each stimulus.json file should also have an allowed JSON sidecar.  The stimulus.tsv file sounds like a timeseries.  Also, it sounds from the description that you would have to provide an annotation for each frame, but what is annotated should be arbitrary.

Reply:
Agreed. Currently, we are not using any (e.g. `the-present-LogLumRatio_stimulus.tsv`). This should change to something like `the-present_annot-LogLumRati_stimulus.tsv`. I'll make the change.